### PR TITLE
Entity split task

### PR DESCRIPTION
**Pull request by xavluiz**
this updates the tracker to allow the dom_event to use the 3 entities: event_meta, plugin, and time_info.

I'll update the tracker info in notion but let me know if the names are ok or any other suggestions. thanks for the feedback

### CODE REVIEW

**Comment by mmclead** [comment]
a folder structure of :
```
src
  events
    - dom_event.ts
    - codetime.ts
    - editor_action.ts
  entities
    - project.ts
    - repo.ts
    - etc..
```
might be better. would fit the mental model of the JSON schema

**Comment by xavluiz** [comment]
good call, will update the structure

File: src/entities/user.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// The user entity
```
Comment (mmclead) [inline src/entities/user.ts:1]: This entity gets created by the API enrichment after looking up the JWT.  So we won't need it in the tracker.

File: src/entities/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// The plugin entity
+export interface Plugin {
```
Comment (mmclead) [inline src/entities/plugin.ts:2]: I think the naming convention would be a bit nicer to use if we called the interfaces `PluginInterface` here and in the class definition had:
```export class Plugin implements PluginInterface```
That way we keep the names of the objects to be just the object itself.

File: src/entities/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  public plugin_name: string;
+
+  constructor(params: Plugin) {
+    this.plugin_id = params.plugin_id;
+    this.plugin_version = params.plugin_version;
+    this.plugin_name = params.plugin_name;
+  }
+}
+
+export async function buildPluginPayload(plugin: Plugin) {
```
Comment (mmclead) [inline src/entities/plugin.ts:20]: Could we make this an "instance method" (sorry, Ruby nomenclature).

Like 
```
buildPayload() {
  return {
     schema: "iglu:com.software/plugin/jsonschema/1-0-0",
     data: {
       ...this
     }
}
```

So when its being used we can do:
```
const plugin: Plugin = new Plugin(params);   // if we change the interface and class names to not have the Impl
const _pluginPayload = await plugin.buildPayload();
```

**Comment by mmclead** [comment]
a few naming changes.    But the structure looks really nice

File: src/entities/user.ts (hunk 1 above)
Comment (xavluiz) [reply src/entities/user.ts:1]: ok, cool

File: src/entities/plugin.ts (hunk 2 above)
Comment (xavluiz) [reply src/entities/plugin.ts:2]: that's fine, but as you know java convention is the Impl way, but will update

File: src/entities/plugin.ts (hunk 3 above)
Comment (xavluiz) [reply src/entities/plugin.ts:20]: yes, totally agree, will update

**Comment by mmclead** [comment]
Looks good to me!

Follow up PR for testing the hashing util: https://github.com/swdotcom/swdc-tracker/pull/7

**Pull request merged**